Editor: add links to theme help in sidebar

timmyc opened the pull request:
This branch adds in some new links to the editor sidebar to provide quick access to theme setup info ( discussion pNEWy-860-p2 ).

__Link in Editor Sidebar__
<img width="256" alt="new_post_ _fly_fishers_place_ _wordpress_com" src="https://cloud.githubusercontent.com/assets/22080/18149994/ab5d47e2-6f98-11e6-9288-6bc55e95bbcc.png">

__Link in Page Editor Sidebar__
<img width="260" alt="new_page_ _fly_fishers_place_ _wordpress_com" src="https://cloud.githubusercontent.com/assets/22080/18150013/ce59da3a-6f98-11e6-8694-a2a496a210aa.png">

__To Test__
The links should be shown as above when using the editor on a WPCOM site ( links hidden on JP sites ).  The Page editor link requires the theme to support page templates - if the theme does not support it, the link is not rendered.

- Open the page editor, verify the link is shown
- Click on the help link.  The link should open the theme showcase page for the given site.  If the active theme on the site is a premium theme, the page should be `http://calypso.localhost:3000/theme/$themeSlug/setup/$siteSlug` - for non-premium themes `http://calypso.localhost:3000/theme/$themeSlug/$siteSlug`

/cc @karmatosed not sure if you want to work on the styling here at all.

Test live: https://calypso.live/?branch=add/editor/theme-help-links

mtias commented:
Can we use compact borderless button with help gridicon for these to standardize? Like:

![image](https://cloud.githubusercontent.com/assets/548849/18165988/cdd7ea4c-7048-11e6-8040-ba3427630548.png)

karmatosed commented:
@mtias interesting idea, I sort of like the explicit nature of the text. It's not help per say it's documentation. I do wonder in this context if we had a Help link the user would expect support or something like that. I'm open to exploring it, but I do feel that maybe in this case as the issue is the documentation being hidden, this explicit calling makes sense.

timmyc commented:
@mtias I had the same thought myself - utilize the existing pattern that we have in the editor sidebar:

![_2__new_post_ _trout_bummin_ _wordpress_com](https://cloud.githubusercontent.com/assets/22080/18176591/11c22cb8-702b-11e6-891b-b7096c0e71ff.png)

karmatosed commented:
I would really worry about just an icon like that, with no word context. I'm not really sure that will solve the issue. However, we could user test. The problem which this is all trying to fix is users getting absolutely lost at that point in more complex themes.

Is the reason a text link isn't wanted because of some visual pattern? If that's the case, maybe we can find some middle ground.

timmyc commented:
> The problem which this is all trying to fix is users getting absolutely lost at that point in more complex themes.

I totally agree here.  Having it behind an (i) or an (?) puts the help 2 clicks away.  I'm totally fine with launching as text and tracking how often the links are clicked.  Then circling back and changing it up as needed.

mtias commented:
I'm a bit worried with how much information and text treatments are packed in the sidebar, sometimes for a similar purpose (like the existing (i)s). Could we look at something like guided-tours for this instead?

Also, if we used the (?) with the borderless compact button I'm not saying to switch the text to just "help", we can keep more specific text of course, but it will have some more familiarity:

![image](https://cloud.githubusercontent.com/assets/548849/18177044/a146158c-7078-11e6-88ad-c4e93757712c.png)

karmatosed commented:
I'm not sure we need something like guided tours. Lets maybe add some context to this with some links you may not be aware of @mtias:

https://themedevp2.wordpress.com/2016/07/06/how-can-we-make-theme-documentation-more-obvious/ is where this all started. The problem is that theme documentation isn't being obvious.

First our approach was this:
https://themedevp2.files.wordpress.com/2016/07/yourtheme.png

We eventually decided on a v2 after @folletto and @melchoyce gave feedback:
https://themedevp2.files.wordpress.com/2016/08/iteration.png

There are a few things important with whatever solution:
- We get something there, we've had a lot of iterations :)
- We have something that doesn't require guess work
- We have something that is similar enough to what we have in Customizer (we have that being worked on), so they at least feel the same.

Hopefully context helps frame some of my responses.

timmyc commented:
To compare the options, in 9e85284 I have switched it over to the compact button with full verbiage:

![new_post_ _trout_bummin_ _wordpress_com](https://cloud.githubusercontent.com/assets/22080/18185243/88f6d830-7051-11e6-8817-a485177401be.png)

mtias commented:
@karmatosed thanks for the screenshots. (The font size seems more balanced in your screenshots.) The reason why we have the info-popover component was to help and explain certain features without getting in the way (and vertical space) of people who don't care about it, that's why I thought this matched the same behaviour and would be consistent with the editor sidebar:

https://wpcalypso.wordpress.com/devdocs/design/info-popover

However, if you want to test these more prominent links, what about about it from...

![image](https://cloud.githubusercontent.com/assets/548849/18247127/d7e4cf8e-7370-11e6-95cf-0fcace133fc2.png)

... to below the format radio selects?

timmyc commented:
rebased.  @karmatosed let me know which path you'd like to take with @mtias feedback above.

mtias commented:
Also, it's fine to get this in if we want to keep discussing the design approach.

karmatosed commented:
Let me just check I've got the right one, I think lets test the prominent links. We totally can adjust after and discuss the design approach. I'd love that actually @mtias as I think we can bring others into the conversation.

karmatosed commented:
@timmyc hi there, I wanted to kick this back off again after the GM. Are we able to go the prominent links route?

timmyc commented:
@karmatosed I moved back to the prominent link location, reduced the size to 12px, and added a simple mc stat bump to track usage:

<img width="259" alt="new_post_ _testingtimmy2_wordpress_com_ _wordpress_com" src="https://cloud.githubusercontent.com/assets/22080/19093217/17759482-8a3f-11e6-8d80-6554f41688fc.png">

karmatosed commented:
@timmyc I did some testing and this seems good to me. Thanks for sorting that. I do feel this is a great first iteration and really does bring a lot of benefit. How shall we proceed? 

@mtias or are you happy with what we have now in this patch, or would you like to explore more?

mtias commented:
Sounds good to me. I think we should move on to an issue to discuss more organic help flows that apply across all sections (maybe using guided tours that you can invoke from the (?) icon in the bottom of the sidebar).

cc @rickybanister and @mrheston

karmatosed commented:
@mtias would you recommend merging this to get the gain of it existing and then do that issue?

mtias commented:
Yes.

karmatosed commented:
Great, @timmyc sounds like we can go ahead and merge. Should we do some form of wider testing or is this low enough impact to just get in?

timmyc commented:
I think this is small enough to just merge in @karmatosed - I'll do so shortly.

karmatosed commented:
@timmyc thanks for the merge and all your work on this <3
